check-log/lib/check-log.go

```diff
@@ -309,11 +318,16 @@ func (opts *logOpts) searchReader(rdr io.Reader) (warnNum, critNum, readBytes in
 }
 
 func (opts *logOpts) match(line string) (bool, []string) {
-	pReg := opts.patternReg
-	eReg := opts.excludeReg
-
-	matches := pReg.FindStringSubmatch(line)
-	matched := len(matches) > 0 && (eReg == nil || !eReg.MatchString(line))
+	matched := true
+	var matches []string
+	for _, pReg := range opts.patternReg {
+		if matched {
+			eReg := opts.excludeReg
+
+			matches = pReg.FindStringSubmatch(line)
+			matched = len(matches) > 0 && (eReg == nil || !eReg.MatchString(line))
+		}
```

Comment:
Thanks for review!
In this case, it is aimed to detect matching all patterns in `opts.patternReg`, so the correct code is below?


```go
for _, pReg := range opts.patternReg {
    eReg := opts.excludeReg

    matches = pReg.FindStringSubmatch(line)
    if len(matches) == 0 || (eReg != nil && eReg.MatchString(line)) {
        return false, nil
    }
}
return true, matches
```